toast-tk-web-agent/src/main/java/io/toast/tk/agent/ui/ConfigPanel.java

```diff
@@ -412,215 +173,156 @@ private void close() {
-                            public java.security.cert.X509Certificate[] getAcceptedIssuers() {
-                                return null;
-                            }
-                            public void checkClientTrusted(
-                                    java.security.cert.X509Certificate[] certs, String authType) {
-                            }
-                            public void checkServerTrusted(
-                                    java.security.cert.X509Certificate[] certs, String authType) {
-                            }
-                        }
-                };
-                SSLContext sc = SSLContext.getInstance("SSL");
-                sc.init(null, trustAllCerts, new java.security.SecureRandom());
-                URL siteURL = new URL(url);
-                HttpsURLConnection.setDefaultSSLSocketFactory(sc.getSocketFactory());
-                // Create all-trusting host name verifier
-                HostnameVerifier allHostsValid = new HostnameVerifier() {
-                    public boolean verify(String hostname, SSLSession session) {
-                        return true;
-                    }
-                };
-                // Install the all-trusting host verifier
-                HttpsURLConnection.setDefaultHostnameVerifier(allHostsValid);
-
-                HttpsURLConnection connection = (HttpsURLConnection) siteURL.openConnection();
-                connection.setSSLSocketFactory(sc.getSocketFactory());
-                connection.setRequestMethod("GET");
-                connection.connect();
-
-                int code = connection.getResponseCode();
-                if (code == 200) {
-                    result = true;
-                }
-            } catch (Exception e) {
-                result = false;
-            }
-        } catch (Exception e) {
-            result = false;
-        }
-        return result;
-    }
+	private void buildFields() throws IOException{
+		//%% API PANEL %%
+		apiKeyPanel = new SimplePanel(properties, AgentConfigProvider.TOAST_API_KEY,
+				UIMessages.USER_API_KEY, EnumError.APIKEY);
+		boxFields.put(AgentConfigProvider.TOAST_API_KEY, apiKeyPanel);
+
+		//%% PLUGIN PANEL %%
+		pluginPanel = new FileChoosePanel(properties, AgentConfigProvider.TOAST_PLUGIN_DIR,
+				UIMessages.PLUGIN_DIR, EnumError.DIRECTORY);
+		boxFields.put(AgentConfigProvider.TOAST_PLUGIN_DIR, pluginPanel);
+
+		//%% SCRIPTS PANEL %%
+		scriptsPanel = new FileChoosePanel(properties, AgentConfigProvider.TOAST_SCRIPTS_DIR,
+				UIMessages.SCRIPT_DIR, EnumError.DIRECTORY);
+		boxFields.put( AgentConfigProvider.TOAST_SCRIPTS_DIR, scriptsPanel);
+
+		//%% CHROME PANEL %%
+		chromePanel = new FileChoosePanel(properties, AgentConfigProvider.TOAST_CHROMEDRIVER_PATH,
+				UIMessages.CHROME_BIN_PATH, EnumError.FILE);
+		boxFields.put(AgentConfigProvider.TOAST_CHROMEDRIVER_PATH, chromePanel);
+
+		//%% PROXY CHECK BOX %%
+		proxyCheckBox = new JCheckBox(UIMessages.ACTIVATE);
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL') Make the enclosing method "static" or remove this set. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS2696)